src/metaopt/io/database/mongodb.py

```diff
@@ -88,26 +90,16 @@ def unlock(self):
             raise_from(DatabaseError("Could not determine unlock due to connection."), e)
         return unlock
 
-    def insert(self, collection_name, document_dictionaries):
-        """ Inserts a document/multiple documents
-
-        Keyword arguments:
-        collection_name -- the collection name
-        document_dictionaries -- the dictionary/dictionaries that would be inserted
-                                 inside the collection
-        Returns:
-        result -- an instance of InsertManyResult which has inserted_ids and acknowledged
-        """
-
-        dbcollection = self.db[collection_name]
-        result = dbcollection.insert_many(document_dictionaries)
-
-        return result
-
-    def update(self, collection_name, query, update_dictionary,
-               selection, upsert=False, sort=None,
-               return_document=ReturnDocument.BEFORE):
-        """ Finds a document and modify it
+    def write(self, collection_name,
+                    query,
+                    dictionary,
+                    selection,
+                    atomic=False,
+                    upsert=False,
+                    sort=None,
+                    return_document=ReturnDocument.BEFORE,
```

Comment:
I'm suspicious that for MySQL or PostgreSQL the interface might be different, I'll do research for now and get back to you for that.

Reply:
The interface will probably be different but we could handle that inside the method. If we define our own codes for return_document we could use the same interface for all type of databases. It could be something like this inside the abstractdb python file.

```python
class ReturnDocument(object):
    NONE = 0
    BEFORE = 1
    AFTER = 2
```

For simplicity I would only implement NONE for now.